src/expr/src/scalar/func.rs

```diff
@@ -4210,6 +4214,30 @@ fn trim_whitespace<'a>(a: Datum<'a>) -> Datum<'a> {
     Datum::from(a.unwrap_str().trim_matches(' '))
 }
 
+fn right<'a>(a: Datum<'a>, b: Datum<'a>) -> Result<Datum<'a>, EvalError> {
+    let string: &'a str = a.unwrap_str();
+    let n = b.unwrap_int64();
+
+    let mut byte_indices = string.char_indices().map(|(i, _)| i);
```

Comment:
`byte_indices` is a bit misleading here, could you rename this to something like `char_indices` instead? (chars can span more than one byte!)